src/Context/Select2Context.php

```diff
@@ -131,4 +133,24 @@ private function selectValue(DocumentElement $page, $field, $value)
 
         throw new \Exception(sprintf('Value "%s" not found for "%s"', $value, $field));
     }
+
+    /**
+     * Wait the end of fetching Select2 results
+     *
+     * @param int $time Time to wait in seconds
+     */
+    private function waitForLoadingResults($time = 60)
+    {
+        for ($i = 0; $i < $time; $i++) {
+            try {
+                if (!$this->getSession()->getPage()->find('css', '.select2-results__option.loading-results')) {
+                    return true;
+                }
+            } catch (\Exception $e) {
+
+            }
```

Comment:
If the loop is executed `$time` times, the proper exception won't be thrown.

I suggest throwing the exception that way in the `catch` instruction : 
```
if ($i === $time - 1) {
    throw $e;
}
```
(or maybe using `$i <= $time` in the loop declaration to avoid the `$time - 1`)